### PR TITLE
Add remember last cursor position indicators & jump functionality

### PR DESCRIPTION
**Pull request by Myzel394**
This PR adds the config `show_last_positions` which will show the last positions for insert / visual / replace mode and also allows the user to jump to it directly.

This is a feature I've wanted for a long time; I first wanted to create a plugin myself but decided it would be better to directly integrate it with SmoothCursor.

You can configure it like this:

```lua
vim.fn.sign_define('smoothcursor_v', { text = '' })
vim.fn.sign_define('smoothcursor_i', { text = '' })
```

**This PR is not ready yet. Some things do not yet such as any mode except for insert mode being shown - I don't know why this is the case, but I'll need to figure that out first**

### CODE REVIEW

**Comment by Myzel394** [comment]
Note to myself: Store both col and line in `last_positions`.

I'm working on adding a small telescope plugin for this too - hence we need to save both the col and line info :D

**Comment by Myzel394** [comment]
All right, telescope plugin available at https://github.com/Myzel394/telescope-last-positions.nvim. I think this PR is ready for a review. I tested everything manually but there may be some weird edge cases I haven't thought of. I'd propose I use the current dev version a few days in the wild and report back if there are any issues

**Comment by gen740** [comment]
Thank you for your PR. I love this feature and will check it soon!

**Comment by gen740** [comment]
@Myzel394
It seems that it's not functioning correctly on my end.
Could you please check if it works with the following configuration?
```lua
require('smoothcursor').setup {
    show_last_positions = nil,  
}

vim.fn.sign_define('smoothcursor_v', { text = ' ' })
vim.fn.sign_define('smoothcursor_V', { text = '' })
vim.fn.sign_define('smoothcursor_i', { text = '' })
vim.fn.sign_define('smoothcursor_�', { text = '' })
vim.fn.sign_define('smoothcursor_R', { text = '󰊄' })
```

File: README.md [hunk 1]
```diff
@@ -155,3 +161,19 @@ autocmd({ 'ModeChanged' }, {
...
+You can decide which modes should be added. 
+Define the `smoothcursor_<mode>` sign group and set your icon.
+Each `<mode>` is a lowercase letter, e.g., `n` for normal mode, `i` for insert mode, etc.
+
+**example**
+```lua
+vim.fn.sign_define('smoothcursor_v', { text = ' ' })
+vim.fn.sign_define('smoothcursor_V', { text = '' })
+vim.fn.sign_define('smoothcursor_i', { text = '' })
+vim.fn.sign_define('smoothcursor_�', { text = '' })
```
Comment (gen740) [inline README.md:176]: Would it be more reasonable to use `smoothcursor_^v` instead?

File: lua/smoothcursor/callbacks/init.lua [hunk 2]
```diff
@@ -32,7 +31,15 @@ local function unplace_signs(with_timer_stop)
...
-  vim.fn.sign_unplace('*', { buffer = vim.fn.bufname(), id = config.value.cursorID })
+
+  vim.fn.sign_unplace(
+    '*',
+    {
+      buffer = vim.fn.bufname(),
+      id = config.value.cursorID
+    }
+  )
+
```
Comment (gen740) [inline lua/smoothcursor/callbacks/init.lua]: Would you mind formatting the code using `stylua`? It seems the change might not be necessary.

**Comment by Myzel394** [comment]
```lua
require('smoothcursor').setup {
    show_last_positions = nil,  
}
```

well you either need to set `show_last_positions = "enter"` or `show_last_positions = "leave"` - `nil` disables it :D

File: lua/smoothcursor/callbacks/init.lua (hunk 2 above)
Comment (Myzel394) [reply lua/smoothcursor/callbacks/init.lua]: should be formatted now

File: README.md (hunk 1 above)
Comment (Myzel394) [reply README.md:176]: Probably yes. I don't really use this block.. visual.. selection thing - I didn't even know vim had something like this until I searched for the icons :O so yeah, "^v" is probably better. Did you test this? I don't know if "^v" is the actual mode name

**Comment by gen740** [comment]
Got it, but I just realized it's not working with the Neovim HEAD. (May be it is a neovim's bug)

File: README.md (hunk 1 above)
Comment (gen740) [reply README.md:176]: You can check out all the return values with `:h mode()`. According to this document, Visual blockwise is expressed as `CTRL-V`, so maybe `smoothcursor_^V` or `smoothcursor_CTRL_V` would be more appropriate. But of course, you'd need to convert `` to `CTRL_V` in some places.

**Comment by Myzel394** [comment]
> Got it, but I just realized it's not working with the Neovim HEAD

I'm not sure what you mean by "Neovim HEAD" - can you elaborate?

**Comment by gen740** [comment]
This refers to NeoVim's HEAD build, specifically the nightly version of NeoVim.

**Comment by gen740** [comment]
However, by implementing #53, I successfully resolved that issue. Could you merge it into the main branch?

**Comment by Myzel394** [comment]
@gen740 I'll do that later

**Comment by Myzel394** [comment]
All right, everything merged and tested on:

```
NVIM v0.9.4
Build type: Release
LuaJIT 2.1.1700008891
```

**Comment by gen740** [comment]
Thank you for the update. In fact, there are some edge cases (e.g., JumpCursor is not properly displayed during animation). However, would it be alright if I address these issues after merging this?

**Comment by Myzel394** [comment]
> However, would it be alright if I address these issues after merging this?

Sure, if you'd like to take this over, feel free to do so :D

**Pull request merged**